lib/trip_dispatcher.rb

```diff
@@ -27,12 +35,51 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def first_available_driver
+      counter = nil
+      max_time = Time.now()
+      self.drivers.each_with_index do |driver, index|
+        if driver.status == :AVAILABLE 
```

Comment:
It seems that you implemented some of the optionals for wave 3 (nice work!), but I don't see tests corresponding to these rules. In addition, the mix of a post-fix conditional and a standard conditional, nested in another conditional, makes this a little tricky to read. Consider breaking out into helper methods to increase readability.